unit-tests-java8/src/test/java/org/eclipse/collections/test/RichIterableTestCase.java

```diff
@@ -272,6 +272,32 @@ default void RichIterable_getLast()
     }
 
     @Test
+    default void RichIterable_getOnly()
+    {
+        RichIterable<Integer> iterable = this.newWith(3);
+        Integer only = iterable.getOnly();
+        assertThat(only, is(3));
+
+        Iterator<Integer> iterator = iterable.iterator();
+        assertThat(iterator.next(), is(only));
+        assertThat(iterator.hasNext(), is(false));
+    }
+
+    @Test
+    default void RichIterable_getOnly_exception_when_empty()
+    {
+        RichIterable<Object> emptyIterable = this.newWith();
+        assertThrows(IllegalStateException.class, () -> emptyIterable.getOnly());
+    }
+
+    @Test
+    default void RichIterable_getOnly_exception_when_two_items()
+    {
+        RichIterable<Object> iterable = this.newWith(1, 2);
+        assertThrows(IllegalStateException.class, () -> iterable.getOnly());
+    }
```

Comment:
These tests look good. I think all three could be combined into a single method, RichIterable_getOnly(). Are the assertions ever overridden separately?

It would be good to include an assertion to show that this.newWith(1, 1).getOnly() throws, with two copies of the same element for collections that support it.